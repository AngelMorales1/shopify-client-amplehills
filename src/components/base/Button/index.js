import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Link } from 'react-router-dom';
import isExternalLink from 'utils/isExternalLink';

import styles from './Button.scss';

const Button = ({
  className,
  label,
  variant,
  color,
  children,
  disabled,
  onClick,
  minWidth,
  fullWidth,
  type,
  to
}) => {
  const classes = cx(
    className,
    'flex flex-column justify-center items-center',
    styles.Button,
    styles[`Button--${variant}`],
    styles[`Button--${color}`],
    {
      w100: fullWidth,
      [styles['Button--disabled']]: disabled
    }
  );

  const linkedComponent = isExternalLink(to) ? (
    <a
      className={cx('text-decoration-none', {
        'events-none': disabled
      })}
      href={to}
      target="_blank"
      rel="noopener"
      onClick={onClick}
    >
      <div className={classes}>{label}</div>
    </a>
  ) : (
    <Link
      className={cx('text-decoration-none', {
        'events-none': disabled
      })}
      to={to}
      onClick={onClick}
    >
      <div className={classes}>{label}</div>
    </Link>
  );

  const button = to ? (
    linkedComponent
  ) : (
    <button type={type} onClick={onClick} className={classes}>
      <div>{children && !label ? children : label}</div>
    </button>
  );

  return button;
};

Button.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
  variant: PropTypes.string,
  color: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  onClick: PropTypes.func,
  fullWidth: PropTypes.bool,
  disabled: PropTypes.bool,
  type: PropTypes.string,
  to: PropTypes.string
};

Button.defaultProps = {
  className: '',
  children: null,
  label: '',
  variant: 'primary',
  color: 'white-denim',
  onClick: () => {},
  fullWidth: false,
  disabled: false,
  type: 'button',
  to: ''
};

export default Button;
