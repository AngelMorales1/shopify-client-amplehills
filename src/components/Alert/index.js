import React from 'react';
import cx from 'classnames';
import get from 'utils/get';
import { Button } from 'components/base';

import styles from './Alert.scss';

const Alert = ({ alert }) => {
  const fields = get(alert, 'fields', {});

  return (
    <div
      className={cx(
        styles['Alert'],
        'z-nav w100 fixed l0 t0 bg-madison-blue bold flex flex-row justify-center items-center'
      )}
    >
      <span className={cx(styles['Alert__text'], 'text-white')}>
        {fields.alertCopy}
      </span>
      <Button
        to={fields.linkUrl}
        variant="style-none"
        className={cx(
          styles['Alert__order-now'],
          styles['Alert__text'],
          'text-peach'
        )}
        label={fields.linkCopy}
      />
    </div>
  );
};

export default Alert;
