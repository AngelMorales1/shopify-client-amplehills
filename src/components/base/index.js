import Button from './Button';
import Image from './Image';
import Radio from './Radio';
import QuantitySelector from './QuantitySelector';
import TextField from './TextField';
import Dropdown from './Dropdown';
import FormFlash from './FormFlash';
import Modal from './Modal';
import Carousel from './Carousel';
import HorizontalCarousel from './HorizontalCarousel';

export {
  Button,
  Image,
  Radio,
  QuantitySelector,
  TextField,
  Dropdown,
  FormFlash,
  Modal,
  Carousel,
  HorizontalCarousel
};
