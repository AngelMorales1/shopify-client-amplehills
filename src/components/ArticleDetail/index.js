import React from 'react';
import get from 'utils/get';
import cx from 'classnames';
import contentfulImgUtil from 'utils/contentfulImgUtil';
import { FacebookShareButton } from 'react-share';
import { Link } from 'react-router-dom';
import articleModel from 'models/articleModel';

import styles from './ArticleDetail.scss';
import { Image } from 'components/base';
import Breadcrumbs from 'components/Breadcrumbs';

const ArticleDetail = ({ article, newsArticles }) => {
  const breadcrumbs = [
    {
      to: '/news',
      label: 'Back to news'
    }
  ];

  const previousArticle = newsArticles[article.index - 1];
  const nextArticle = newsArticles[article.index + 1];
  const articleIsFirstOne = article.index === 0;
  const articleIsLastOne =
    article.index === newsArticles[newsArticles.length - 1].index;

  return (
    <div className={cx(styles['ArticleDetail'])}>
      <Breadcrumbs
        breadcrumbs={breadcrumbs}
        className={cx(
          styles['ArticleDetail__breadcrumbs'],
          'transition-slide-up mx-auto container-width'
        )}
      />
      <div className="flex flex-column items-center form-container-width mx-auto px2">
        <h2
          className={cx(
            styles['ArticleDetail__title'],
            'bold center text-container-width mt3'
          )}
        >
          {article.title}
        </h2>
        <p className="line-item-title text-peach my2">{article.publishedAt}</p>
        <div
          className={cx(styles['ArticleDetail__content'], 'mb3')}
          dangerouslySetInnerHTML={{ __html: article.contentHtml }}
        />
        <div className="self-start block-subheadline my3">
          <p className="uppercase">Date</p>
          <p>{article.publishedAt}</p>
        </div>
        <div className="self-start block-subheadline">
          <p className="uppercase">Written By</p>
          <p>{article.authorName}</p>
        </div>
      </div>
      <div
        className={cx(
          'flex flex-row items-center justify-center mx-auto mt4 px1',
          {
            'container-width': !articleIsFirstOne || !articleIsLastOne,
            'form-container-width': articleIsFirstOne || articleIsLastOne
          }
        )}
      >
        <Link
          to={`/news/${get(previousArticle, 'handle', '')}`}
          className={cx(
            { 'display-none': articleIsFirstOne },
            styles['ArticleDetail__nav-button-container'],
            'relative mx1 flex flex-column xs-hide sm-hide self-stretch'
          )}
        >
          <Image
            className={cx(
              styles['ArticleDetail__previous-icon'],
              'absolute self-start'
            )}
            src="/assets/images/icon-pagination-previous-arrow.svg"
          />
          <p className="uppercase text-peach bold">Previous</p>
          <p
            className={cx(
              styles['ArticleDetail__nav-button-title'],
              'bold text-madison-blue'
            )}
          >
            {get(previousArticle, 'title', '')}
          </p>
        </Link>
        <Link
          to={`/news/${get(nextArticle, 'handle', '')}`}
          className={cx(
            { 'display-none': articleIsLastOne },
            styles['ArticleDetail__nav-button-container'],
            'relative mx1 flex flex-column self-stretch'
          )}
        >
          <p className="uppercase text-peach bold">Next</p>
          <Image
            className={cx(
              styles['ArticleDetail__next-icon'],
              'absolute self-end'
            )}
            src="/assets/images/icon-pagination-next-arrow.svg"
          />
          <p
            className={cx(
              styles['ArticleDetail__nav-button-title'],
              'bold text-madison-blue'
            )}
          >
            {get(nextArticle, 'title', '')}
          </p>
        </Link>
      </div>
    </div>
  );
};

ArticleDetail.propTypes = {
  article: articleModel.propTypes
};

ArticleDetail.defaultProps = {
  article: {}
};

export default ArticleDetail;
