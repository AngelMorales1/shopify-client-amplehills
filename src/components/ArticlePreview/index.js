import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import cx from 'classnames';
import contentfulImgUtil from 'utils/contentfulImgUtil';
import { FacebookShareButton } from 'react-share';
import articleModel from 'models/articleModel';

import styles from './ArticlePreview.scss';
import { Button } from 'components/base';

const ArticlePreview = ({ article, index }) => {
  const thisIsfirstArticle = index === 0;
  const firstTwoCentencesFromContent =
    article.content
      .split('. ')
      .slice(0, 2)
      .join('. ') + '.';

  return (
    <div className="px2">
      <Button variant="style-none" to={`/news/${article.handle}`}>
        <div
          className={cx(
            styles['ArticlePreview__image'],
            'col-12 aspect-4-3 absolute'
          )}
          style={{
            background: `url(${contentfulImgUtil(
              article.image,
              '900'
            )}) no-repeat center`,
            backgroundSize: 'cover'
          }}
        />
      </Button>
      <div
        className={cx(
          styles['ArticlePreview__text-container'],
          'text-container-width'
        )}
      >
        <Button
          className={cx(styles['ArticlePreview__button'])}
          variant="style-none"
          to={`/news/${article.handle}`}
        >
          <h2
            className={cx(
              styles['ArticlePreview__title'],
              'bold my2 text-madison-blue avenir'
            )}
          >
            {article.title}
          </h2>
        </Button>
        <p
          className={cx(
            styles['ArticlePreview__date'],
            'detail text-peach semi-bold mb3'
          )}
        >
          {article.publishedAt}
        </p>
        {thisIsfirstArticle ? (
          <Fragment>
            <p className="block-subheadline">{firstTwoCentencesFromContent}</p>
            <div className="flex flex-row my3">
              <Button
                to={`/news/${article.handle}`}
                className="mr2"
                variant="primary-small"
                color="madison-blue"
                label="Read More"
              />
              <FacebookShareButton
                url={`${get(window, 'location.href', '')}/${article.handle}`}
              >
                <Button
                  variant="primary-small"
                  color="white-madison-blue-border"
                  label="Share"
                />
              </FacebookShareButton>
            </div>
          </Fragment>
        ) : null}
      </div>
    </div>
  );
};

ArticlePreview.propTypes = {
  article: articleModel.propTypes,
  index: PropTypes.number
};

ArticlePreview.defaultProps = {
  article: {},
  index: 0
};

export default ArticlePreview;
