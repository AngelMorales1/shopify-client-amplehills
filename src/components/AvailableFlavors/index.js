import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import get from 'utils/get';
import contentfulImgUtil from 'utils/contentfulImgUtil';
import Global from 'constants/Global';
import locationModel from 'models/locationModel';
import slugify from 'utils/slugify';

import { Button, Image } from 'components/base';
import styles from './AvailableFlavors.scss';

class AvailableFlavors extends Component {
  state = {
    currentBreakpoint: Global.breakpoints.medium.label,
    showAllFlavorOnMobileView: false
  };

  componentDidMount() {
    window.addEventListener('resize', this.updateWindow);
    this.updateWindow();
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindow);
  }

  updateWindow = () => {
    const { small, medium } = Global.breakpoints;
    const currentBreakpoint =
      window.innerWidth <= medium.lowerbound ? small.label : medium.label;

    if (this.state.currentBreakpoint !== currentBreakpoint)
      this.setState({ currentBreakpoint });
  };

  render() {
    const { drip, setRef, location, block, z } = this.props;
    const { small } = Global.breakpoints;
    const fields = get(block, 'fields', {});
    const title = get(fields, 'title', '');
    const backgroundColor = get(fields, 'backgroundColor', 'white');
    const locationId = get(location, 'id', '');
    const isMobileView = this.state.currentBreakpoint === small.label;
    const availableFlavors = get(location, 'availableFlavors', []);
    const selectedAvailableFlavors =
      isMobileView && !this.state.showAllFlavorOnMobileView
        ? availableFlavors.slice(0, 4)
        : availableFlavors;
    const FlavorsLength = availableFlavors.length;

    return (
      <div
        ref={refBlock => setRef(refBlock)}
        style={{ zIndex: z }}
        className={cx(
          { drip: drip },
          styles[`AvailableFlavors--${backgroundColor}`],
          'py4 px3'
        )}
      >
        <div className="mx-auto flex flex-column items-center container-width">
          <h2 className="block-headline center mb3">{title}</h2>
          <div
            className={cx(
              styles['AvailableFlavors__card-container'],
              'flex flex-wrap items-center justify-center'
            )}
          >
            {selectedAvailableFlavors.map((flavor, i) => {
              const fields = get(flavor, 'fields', {});
              const title = get(fields, 'title', '');
              const label = get(fields, 'label', '');
              const image = get(fields, 'image.fields.file.url', '');
              const slug = slugify(title);
              const isLocationSpecial = get(fields, 'locationSpecial', []).find(
                location => {
                  return get(location, 'sys.id', '') === locationId;
                }
              );

              return (
                <Button
                  to={`/flavors/${slug}`}
                  key={get(flavor, 'sys.id', i)}
                  variant="style-none"
                  className={cx(
                    styles['AvailableFlavors__card'],
                    'flex flex-row justify-between items-center bg-white m1'
                  )}
                >
                  <Image
                    className={cx(
                      styles['AvailableFlavors__card-image'],
                      'col-5 mr1'
                    )}
                    src={contentfulImgUtil(image, '200')}
                  />
                  <div className="col-6 ml1">
                    {label &&
                      !isLocationSpecial && (
                        <p className="mb1 carter light detail text-peach">
                          {label}
                        </p>
                      )}
                    {isLocationSpecial && (
                      <p className="mb1 carter light detail text-peach">
                        Location Special
                      </p>
                    )}
                    <p className="small bold text-madison-blue">{title}</p>
                  </div>
                </Button>
              );
            })}
            {isMobileView && FlavorsLength > 4 ? (
              <Button
                className="mt3"
                variant="primary-small"
                color="peach"
                onClick={() =>
                  this.setState({
                    showAllFlavorOnMobileView: !this.state
                      .showAllFlavorOnMobileView
                  })
                }
                label={
                  this.state.showAllFlavorOnMobileView
                    ? 'View Less Flavors'
                    : `View ${FlavorsLength - 4} More Flavors`
                }
              />
            ) : null}
          </div>
        </div>
      </div>
    );
  }
}

AvailableFlavors.propTypes = {
  drip: PropTypes.bool,
  setRef: PropTypes.func,
  location: locationModel.propTypes,
  block: PropTypes.shape({
    title: PropTypes.string
  }),
  z: PropTypes.number
};

AvailableFlavors.defaultProps = {
  drip: false,
  setRef: () => {},
  location: locationModel.default,
  block: {
    title: ''
  },
  z: 1
};

export default AvailableFlavors;
