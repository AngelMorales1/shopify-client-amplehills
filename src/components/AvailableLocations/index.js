import React from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import cx from 'classnames';
import { Button } from 'components/base';

import styles from './AvailableLocations.scss';

const AvailableLocations = ({ flavor, block, z, setRef, drip, upperDrip }) => {
  const availableLocations = get(flavor, 'availableLocations', []);
  const fields = get(block, 'fields', {});
  const blockTitle = get(fields, 'title', '');
  const backgroundColor = get(fields, 'backgroundColor', 'white');

  return (
    <div
      ref={refBlock => setRef(refBlock)}
      style={{ zIndex: z }}
      className={cx(
        styles['AvailableLocations'],
        styles[`AvailableLocations--${backgroundColor}`],
        {
          drip: drip,
          'upper-drip': upperDrip
        }
      )}
    >
      <h2 className="block-headline center mb3">{blockTitle}</h2>
      <div className="w100 flex justify-center items-center">
        <div
          className={cx(
            styles['AvailableLocations__button-container'],
            'flex flex-wrap justify-center items-center'
          )}
        >
          {availableLocations.map(location => {
            const fields = get(location, 'fields', {});
            const locationTitle = get(fields, 'title', '');
            const slug = get(fields, 'slug', '');
            const id = get(location, 'sys.id', '');

            return (
              <Button
                key={id}
                className={cx(
                  styles['AvailableLocations__button'],
                  'uppercase inline-flex m1'
                )}
                variant="primary-small"
                color="madison-blue"
                label={locationTitle}
                to={`/location/${slug}`}
              />
            );
          })}
        </div>
      </div>
    </div>
  );
};

AvailableLocations.propTypes = {
  flavor: PropTypes.shape({
    availableLocations: PropTypes.array,
    contentBlocks: PropTypes.array,
    id: PropTypes.string,
    image: PropTypes.string,
    label: PropTypes.string,
    slug: PropTypes.string,
    title: PropTypes.string
  }),
  z: PropTypes.number,
  setRef: PropTypes.func,
  block: PropTypes.shape({
    fields: PropTypes.shape({
      contentType: PropTypes.string,
      drip: PropTypes.bool,
      title: PropTypes.string
    }),
    sys: PropTypes.shape({
      id: PropTypes.string
    })
  })
};

AvailableLocations.defaultProps = {
  flavor: {
    availableLocations: [],
    contentBlocks: [],
    id: '',
    image: '',
    label: '',
    slug: '',
    title: ''
  },
  z: 1,
  setRef: () => {},
  block: {
    fields: {
      contentType: '',
      drip: false,
      title: ''
    },
    sys: {
      id: ''
    }
  }
};

export default AvailableLocations;
