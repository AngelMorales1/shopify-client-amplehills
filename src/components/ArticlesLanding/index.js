import React, { Component } from 'react';
import PropTypes from 'prop-types';
import get from 'utils/get';
import getPagination from 'utils/getPagination';
import cx from 'classnames';
import articleModel from 'models/articleModel';
import { Image, Button } from 'components/base';
import Global from 'constants/Global';

import styles from './ArticlesLanding.scss';
import ArticlePreview from 'components/ArticlePreview';
import RecentArticle from 'components/RecentArticle';

class ArticlesLanding extends Component {
  state = {
    selectedPage: 1,
    currentBreakpoint: Global.breakpoints.small.label
  };

  componentDidMount() {
    window.addEventListener('resize', this.updateWindow);

    this.updateWindow();
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindow);
  }

  updateWindow = () => {
    const { small, large } = Global.breakpoints;
    const currentBreakpoint =
      window.innerWidth <= large.lowerbound ? small.label : large.label;

    if (this.state.currentBreakpoint !== currentBreakpoint) {
      this.setState({ currentBreakpoint });
    }
  };

  handleTagButtonClick = tag => {
    const { removeSelectedTag, addSelectedTag } = get(
      this,
      'props.actions',
      {}
    );
    this.setState({ selectedPage: 1 });

    if (this.props.selectedTags.indexOf(tag) > -1) {
      return removeSelectedTag(tag);
    }

    return addSelectedTag(tag);
  };

  render() {
    const articlesByTags = get(this, 'props.articlesByTags', []);
    const articles = articlesByTags.length
      ? articlesByTags
      : get(this, 'props.articles', []);
    const tags = get(this, 'props.tags', []);
    const articleLength = articles.length;
    const { large } = Global.breakpoints;
    const pageOffset = this.state.currentBreakpoint === large.label ? 2 : 1;
    const { pagination, totalPages } = getPagination(
      articleLength,
      5,
      pageOffset,
      this.state.selectedPage
    );
    const currentPage = this.state.selectedPage;

    return (
      <div
        className={cx(
          styles['ArticlesLanding'],
          'flex justify-between py4 px2'
        )}
      >
        <div className="col-12 md-col-3 px2">
          <h2 className="callout mb2">Recent Articles</h2>
          <div className="flex flex-column mb3">
            {articles
              .slice(0, 4)
              .map(article => (
                <RecentArticle key={article.id} article={article} />
              ))}
          </div>
        </div>
        <div
          className={cx(
            styles['ArticlesLanding__article-preview-container'],
            'col-12 md-col-6'
          )}
        >
          {articles
            .slice(this.state.selectedPage * 5 - 5, this.state.selectedPage * 5)
            .map(article => (
              <ArticlePreview key={article.id} article={article} />
            ))}
          <div className="w100 flex flex-row justify-end flex-wrap items-center px2">
            <Button
              className={cx(
                { 'display-none': currentPage === 1 || totalPages <= 1 },
                'mr1 transition'
              )}
              onClick={() => this.setState({ selectedPage: currentPage - 1 })}
              variant="style-none"
            >
              <Image
                className={cx(
                  styles['ArticlesLanding__pagination-image'],
                  'mr1'
                )}
                src="/assets/images/icon-pagination-previous-arrow.svg"
              />
              <p className="copy text-peach bold xs-hide sm-hide">Previous</p>
            </Button>
            <Button
              className={cx(
                { 'display-none': pagination.includes(1) || totalPages <= 1 },
                'copy text-peach bold'
              )}
              onClick={() => this.setState({ selectedPage: 1 })}
              variant="style-none"
              label="1"
            />
            <p
              className={cx(
                {
                  'display-none':
                    pagination.includes(1) ||
                    pagination.includes(2) ||
                    totalPages <= 1
                },
                'copy text-peach bold mx1'
              )}
            >
              ...
            </p>
            {pagination.map(page => (
              <Button
                className={cx(
                  styles['ArticlesLanding__pagination-number'],
                  'copy text-peach bold'
                )}
                key={page}
                onClick={() => this.setState({ selectedPage: page })}
                variant="style-none"
                label={`${page}`}
              />
            ))}
            <p
              className={cx(
                {
                  'display-none':
                    pagination.includes(totalPages) ||
                    pagination.includes(totalPages - 1) ||
                    totalPages <= 1
                },
                'copy text-peach bold mx1'
              )}
            >
              ...
            </p>
            <Button
              className={cx(
                {
                  'display-none':
                    pagination.includes(totalPages) || totalPages <= 1
                },
                'copy text-peach bold'
              )}
              onClick={() => this.setState({ selectedPage: totalPages })}
              variant="style-none"
              label={`${totalPages}`}
            />
            <Button
              className={cx(
                {
                  'display-none': currentPage === totalPages || totalPages <= 1
                },
                'ml1 transition'
              )}
              onClick={() => this.setState({ selectedPage: currentPage + 1 })}
              variant="style-none"
            >
              <p className="copy text-peach bold xs-hide sm-hide">Next</p>
              <Image
                className={cx(
                  styles['ArticlesLanding__pagination-image'],
                  'ml1'
                )}
                src="/assets/images/icon-pagination-next-arrow.svg"
              />
            </Button>
          </div>
        </div>
        <div
          className={cx(
            styles['ArticlesLanding__line'],
            'mb3 px2 mx2 md-hide lg-hide'
          )}
        />
        <div className="col-12 md-col-3 px2">
          <h2 className="callout mb2">Tags</h2>
          {tags.map(tag => {
            const buttonIsSelected =
              get(this, 'props.selectedTags', []).indexOf(tag) > -1;

            return (
              <Button
                onClick={() => this.handleTagButtonClick(tag)}
                key={tag}
                className="uppercase mb1 mr1"
                label={tag}
                color="green"
                variant={cx({
                  tag: !buttonIsSelected,
                  'selected-tag': buttonIsSelected
                })}
              />
            );
          })}
        </div>
      </div>
    );
  }
}

ArticlesLanding.propTypes = {
  articles: PropTypes.arrayOf(articleModel.propTypes),
  tags: PropTypes.array
};

ArticlesLanding.defaultProps = {
  articles: [],
  tags: []
};

export default ArticlesLanding;
