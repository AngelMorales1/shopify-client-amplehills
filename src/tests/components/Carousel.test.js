import React from 'react';
import { shallow } from 'enzyme';

import { Carousel } from 'components/base';

it('renders without data', () => {
  const component = shallow(<Carousel />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <Carousel
      className="foo"
      showArrows={true}
      showDots={true}
      arrowPreviousImage="/assets/images/icon-circle-left-arrow.svg"
      arrowNextImage="/assets/images/icon-circle-right-arrow.svg"
      onChange={() => console.log('foo')}
      index={0}
    />
  );

  expect(component).toMatchSnapshot();
});
