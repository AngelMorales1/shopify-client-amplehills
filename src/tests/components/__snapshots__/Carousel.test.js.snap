// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Carousel
    arrowNextImage="/assets/images/icon-circle-right-arrow.svg"
    arrowPreviousImage="/assets/images/icon-circle-left-arrow.svg"
    className="foo"
    dotColor=""
    index={0}
    onChange={[Function]}
    showArrows={true}
    showDots={true}
    showOnImage={false}
    sliderClasses=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Carousel
        afterSlide={[Function]}
        autoGenerateStyleTag={true}
        autoplay={false}
        autoplayInterval={3000}
        beforeSlide={[Function]}
        cellAlign="left"
        cellSpacing={0}
        className="w100 flex pb4"
        dragging={true}
        easing="easeCircleOut"
        edgeEasing="easeElasticOut"
        frameOverflow="hidden"
        framePadding="0px"
        heightMode="max"
        onResize={[Function]}
        pauseOnHover={true}
        renderBottomCenterControls={[Function]}
        renderCenterLeftControls={[Function]}
        renderCenterRightControls={[Function]}
        slideIndex={0}
        slideWidth={1}
        slidesToScroll={1}
        slidesToShow={1}
        speed={500}
        style={Object {}}
        swiping={true}
        transitionMode="scroll"
        vertical={false}
        width="100%"
        withoutControls={false}
        wrapAround={false}
/>,
      "className": "foo flex flex-column justify-center transition",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "afterSlide": [Function],
        "autoGenerateStyleTag": true,
        "autoplay": false,
        "autoplayInterval": 3000,
        "beforeSlide": [Function],
        "cellAlign": "left",
        "cellSpacing": 0,
        "children": null,
        "className": "w100 flex pb4",
        "dragging": true,
        "easing": "easeCircleOut",
        "edgeEasing": "easeElasticOut",
        "frameOverflow": "hidden",
        "framePadding": "0px",
        "heightMode": "max",
        "onResize": [Function],
        "pauseOnHover": true,
        "renderBottomCenterControls": [Function],
        "renderCenterLeftControls": [Function],
        "renderCenterRightControls": [Function],
        "slideIndex": 0,
        "slideWidth": 1,
        "slidesToScroll": 1,
        "slidesToShow": 1,
        "speed": 500,
        "style": Object {},
        "swiping": true,
        "transitionMode": "scroll",
        "vertical": false,
        "width": "100%",
        "withoutControls": false,
        "wrapAround": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Carousel
          afterSlide={[Function]}
          autoGenerateStyleTag={true}
          autoplay={false}
          autoplayInterval={3000}
          beforeSlide={[Function]}
          cellAlign="left"
          cellSpacing={0}
          className="w100 flex pb4"
          dragging={true}
          easing="easeCircleOut"
          edgeEasing="easeElasticOut"
          frameOverflow="hidden"
          framePadding="0px"
          heightMode="max"
          onResize={[Function]}
          pauseOnHover={true}
          renderBottomCenterControls={[Function]}
          renderCenterLeftControls={[Function]}
          renderCenterRightControls={[Function]}
          slideIndex={0}
          slideWidth={1}
          slidesToScroll={1}
          slidesToShow={1}
          speed={500}
          style={Object {}}
          swiping={true}
          transitionMode="scroll"
          vertical={false}
          width="100%"
          withoutControls={false}
          wrapAround={false}
/>,
        "className": "foo flex flex-column justify-center transition",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "afterSlide": [Function],
          "autoGenerateStyleTag": true,
          "autoplay": false,
          "autoplayInterval": 3000,
          "beforeSlide": [Function],
          "cellAlign": "left",
          "cellSpacing": 0,
          "children": null,
          "className": "w100 flex pb4",
          "dragging": true,
          "easing": "easeCircleOut",
          "edgeEasing": "easeElasticOut",
          "frameOverflow": "hidden",
          "framePadding": "0px",
          "heightMode": "max",
          "onResize": [Function],
          "pauseOnHover": true,
          "renderBottomCenterControls": [Function],
          "renderCenterLeftControls": [Function],
          "renderCenterRightControls": [Function],
          "slideIndex": 0,
          "slideWidth": 1,
          "slidesToScroll": 1,
          "slidesToShow": 1,
          "speed": 500,
          "style": Object {},
          "swiping": true,
          "transitionMode": "scroll",
          "vertical": false,
          "width": "100%",
          "withoutControls": false,
          "wrapAround": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Carousel
    arrowNextImage="/assets/images/icon-circle-right-arrow.svg"
    arrowPreviousImage="/assets/images/icon-circle-left-arrow.svg"
    className="w100"
    dotColor=""
    index={0}
    onChange={[Function]}
    showArrows={true}
    showDots={true}
    showOnImage={false}
    sliderClasses=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Carousel
        afterSlide={[Function]}
        autoGenerateStyleTag={true}
        autoplay={false}
        autoplayInterval={3000}
        beforeSlide={[Function]}
        cellAlign="left"
        cellSpacing={0}
        className="w100 flex pb4"
        dragging={true}
        easing="easeCircleOut"
        edgeEasing="easeElasticOut"
        frameOverflow="hidden"
        framePadding="0px"
        heightMode="max"
        onResize={[Function]}
        pauseOnHover={true}
        renderBottomCenterControls={[Function]}
        renderCenterLeftControls={[Function]}
        renderCenterRightControls={[Function]}
        slideIndex={0}
        slideWidth={1}
        slidesToScroll={1}
        slidesToShow={1}
        speed={500}
        style={Object {}}
        swiping={true}
        transitionMode="scroll"
        vertical={false}
        width="100%"
        withoutControls={false}
        wrapAround={false}
/>,
      "className": "w100 flex flex-column justify-center transition",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "afterSlide": [Function],
        "autoGenerateStyleTag": true,
        "autoplay": false,
        "autoplayInterval": 3000,
        "beforeSlide": [Function],
        "cellAlign": "left",
        "cellSpacing": 0,
        "children": null,
        "className": "w100 flex pb4",
        "dragging": true,
        "easing": "easeCircleOut",
        "edgeEasing": "easeElasticOut",
        "frameOverflow": "hidden",
        "framePadding": "0px",
        "heightMode": "max",
        "onResize": [Function],
        "pauseOnHover": true,
        "renderBottomCenterControls": [Function],
        "renderCenterLeftControls": [Function],
        "renderCenterRightControls": [Function],
        "slideIndex": 0,
        "slideWidth": 1,
        "slidesToScroll": 1,
        "slidesToShow": 1,
        "speed": 500,
        "style": Object {},
        "swiping": true,
        "transitionMode": "scroll",
        "vertical": false,
        "width": "100%",
        "withoutControls": false,
        "wrapAround": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Carousel
          afterSlide={[Function]}
          autoGenerateStyleTag={true}
          autoplay={false}
          autoplayInterval={3000}
          beforeSlide={[Function]}
          cellAlign="left"
          cellSpacing={0}
          className="w100 flex pb4"
          dragging={true}
          easing="easeCircleOut"
          edgeEasing="easeElasticOut"
          frameOverflow="hidden"
          framePadding="0px"
          heightMode="max"
          onResize={[Function]}
          pauseOnHover={true}
          renderBottomCenterControls={[Function]}
          renderCenterLeftControls={[Function]}
          renderCenterRightControls={[Function]}
          slideIndex={0}
          slideWidth={1}
          slidesToScroll={1}
          slidesToShow={1}
          speed={500}
          style={Object {}}
          swiping={true}
          transitionMode="scroll"
          vertical={false}
          width="100%"
          withoutControls={false}
          wrapAround={false}
/>,
        "className": "w100 flex flex-column justify-center transition",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "afterSlide": [Function],
          "autoGenerateStyleTag": true,
          "autoplay": false,
          "autoplayInterval": 3000,
          "beforeSlide": [Function],
          "cellAlign": "left",
          "cellSpacing": 0,
          "children": null,
          "className": "w100 flex pb4",
          "dragging": true,
          "easing": "easeCircleOut",
          "edgeEasing": "easeElasticOut",
          "frameOverflow": "hidden",
          "framePadding": "0px",
          "heightMode": "max",
          "onResize": [Function],
          "pauseOnHover": true,
          "renderBottomCenterControls": [Function],
          "renderCenterLeftControls": [Function],
          "renderCenterRightControls": [Function],
          "slideIndex": 0,
          "slideWidth": 1,
          "slidesToScroll": 1,
          "slidesToShow": 1,
          "speed": 500,
          "style": Object {},
          "swiping": true,
          "transitionMode": "scroll",
          "vertical": false,
          "width": "100%",
          "withoutControls": false,
          "wrapAround": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
